src/adaptive_graph_of_thoughts/domain/services/neo4j_utils.py

```
@@ -218,28 +228,23 @@ def _transaction_work(tx: Transaction) -> list[Record]:
-    for key in updates:
-        if not key.replace('_', '').replace('-', '').isalnum():
-            raise ValueError(f"Invalid property name: {key}. Property names must be alphanumeric with underscores/hyphens only.")
 
-    set_clause = ", ".join(f"n.{k} = ${k}" for k in updates)
-    query = f"MATCH (n) WHERE id(n) = $id SET {set_clause} RETURN n"
+async def update_node(node_id: str, updates: dict[str, Any]) -> list[Record]:
     try:
-        params = {"id": int(node_id), **updates}
+        node_id_int = int(node_id)
     except ValueError:
         raise ValueError(f"Invalid node_id: {node_id}. Must be a valid integer.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)